logstash-core/src/main/java/org/logstash/execution/Input.java

```diff
@@ -0,0 +1,78 @@
+package org.logstash.execution;
+
+import java.util.Collections;
+import java.util.Scanner;
+import java.util.concurrent.CountDownLatch;
+
+/**
+ * A Logstash Pipeline Input pushes to a {@link QueueWriter}.
+ */
+public interface Input extends AutoCloseable {
+
+    /**
+     * Start pushing {@link org.logstash.Event} to given {@link QueueWriter}.
+     * @param writer Queue Writer to Push to
+     */
+    void start(QueueWriter writer);
+
+    /**
+     * Stop the input.
+     * Stopping happens asynchronously, use {@link #awaitStop()} to make sure that the input has
+     * finished.
+     */
+    void stop();
+
+    /**
+     * Blocks until the input execution has finished.
+     * @throws InterruptedException On Interrupt
+     */
+    void awaitStop() throws InterruptedException;
+
+    @LogstashPlugin(name = "stream")
+    final class StreamInput implements Input {
+
+        private Scanner inpt;
+
+        private final CountDownLatch done = new CountDownLatch(1);
+
+        private volatile boolean stopped;
+
+        /**
+         * Required Constructor Signature only taking a {@link LsConfiguration}.
+         * @param configuration Logstash Configuration
+         * @param context Logstash Context
+         */
+        public StreamInput(final LsConfiguration configuration, final LsContext context) {
+            // Do whatever
+        }
+
+        @Override
+        public void start(final QueueWriter writer) {
+            inpt = new Scanner(System.in, "\n");
+            try {
+                while (!stopped && inpt.hasNext()) {
+                    final String message = inpt.next();
+                    writer.push(Collections.singletonMap("message", message));
```

Comment:
Can you talk me through why the writer writes `Map` objects, not `Event` objects? Was that more efficient given that a `Map` is more lightweight?

Reply:
@andrewvc yea that was just an idea for optimizing things a little more. It's kinda annoying that we have all these `Event`that don't hold any state. For filters and outputs I can see the value in having `Event` around for serialization and such, but for `Input`s it's just a waste of memory to already create `Event` when all inputs in fact simply create `Map<String, Object>`.